util/desensitize/desensitize.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package desensitize
+
+import (
+	"github.com/pingcap/tidb/kv"
+)
+
+// Error is used to desensitize error message.
+func Error(err error, desensitize bool) error {
```

Comment:
Rather than pass in the bool in the caller can check the bool themselves. This both avoid some overhead and decreases the cyclomatic complexity of the code.

Reply:
I would call this KeyError unless it is going to redact multiple errors. And actually I would put it in the kv module. I think this function could be useful from there for tests. But it also organizes the code such that the kv module knows how many args to give to the error.

Reply:
Great, done.